WordPress/Classes/ViewRelated/Themes/ThemeBrowserViewController.swift

```diff
@@ -1,6 +1,46 @@
 import Foundation
 
-@objc public class ThemeBrowserViewController : UICollectionViewController, UICollectionViewDelegateFlowLayout, NSFetchedResultsControllerDelegate, UISearchBarDelegate, WPContentSyncHelperDelegate {
+public enum ThemeType {
+    case All
+    case Free
+    case Premium
+    
+    static let types = [All, Free, Premium]
+
+    var title: String {
+        switch self {
+        case .All:
+            return NSLocalizedString("All", comment: "Browse all themes selection title")
+        case .Free:
+            return NSLocalizedString("Free", comment: "Browse free themes selection title")
+        case .Premium:
+            return NSLocalizedString("Premium", comment: "Browse premium themes selection title")
+        }
+    }
+    
+    var predicate: NSPredicate? {
+        switch self {
+        case .All:
+            return nil
+        case .Free:
+            return NSPredicate(format: "premium == 0")
+        case .Premium:
+            return NSPredicate(format: "premium == 1")
+        }
+    }
+}
+
+public protocol ThemePresenter: class {
```

Comment:
You need `: class` to declare a weak delegate of that protocol type, to ensure it can't have an enum or struct value type assigned, just a [class instance reference type](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/Protocols.html#//apple_ref/doc/uid/TP40014097-CH25-ID281).